plugins/BEdita/API/src/Controller/LoginController.php

```diff
@@ -86,22 +88,81 @@ public function initialize()
             $this->Auth->setConfig('authenticate', $authenticationComponents, false);
         }
 
+        if ($this->request->getParam('action') === 'optout') {
+            $this->Auth->setConfig('loginAction', ['_name' => 'api:login:optout']);
+        }
+
         if ($this->request->getParam('action') === 'change') {
             $this->Auth->getAuthorize('BEdita/API.Endpoint')->setConfig('defaultAuthorized', true);
         }
     }
 
     /**
-     * Login action use cases:
+     * Login action via user identification with classic username/password, OTP, Oauth2 or 2FA.
+     * See `identify` method for more details.
+     *
+     * @return void
+     * @throws \Cake\Http\Exception\UnauthorizedException Throws an exception if user credentials are invalid or acces is not authorized
+     */
+    public function login(): void
+    {
+        $result = $this->identify();
+        // Check if result contains only an authorization code (OTP & 2FA use cases)
+        if (!empty($result['authorization_code']) && count($result) === 1) {
+            $meta = ['authorization_code' => $result['authorization_code']];
+        } else {
+            $result = $this->reducedUserData($result);
+            $meta = $this->jwtTokens($result);
+        }
+
+        $this->set('_serialize', []);
+        $this->set('_meta', $meta);
+    }
```

Comment:
minor... function 'paths' could be reduced... i.e.:
```
    public function login(): void
    {
        $this->set('_serialize', []);
        $result = $this->identify();
        // Check if result contains only an authorization code (OTP & 2FA use cases)
        if (!empty($result['authorization_code']) && count($result) === 1) {
            $this->set('_meta', ['authorization_code' => $result['authorization_code']]);
            
            return;
        }
        $result = $this->reducedUserData($result);
        $meta = $this->jwtTokens($result);
        $this->set('_meta', $meta);
    }
```

Reply:
done